src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/IO/TdsStream.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Data.SqlClientX.IO
+{
+    /// <summary>
+    /// A Stream abstraction over the TDS protocol.
+    /// The stream can be used to read and write TDS messages on a 
+    /// SQL Server physical connection.
+    /// </summary>
+    internal class TdsStream : Stream
```

Comment:
Is the stream disposable? Can it implement DisposeAsync pattern?

Reply:
Good catch. It should